client/src/components/Post/Post.tsx

```diff
@@ -4,48 +4,90 @@ import {RouteComponentProps, withRouter} from "react-router-dom";
+                {id: 0, editDate: null, username: 'Jhon Mark', content: 'This is the body for a comment post. asd asd asd asd sad asd as das das dasda sdasd asdasdasd asdasd asdas dasdasd asda sdas dasdasd asdas dasd asd as dasd asd asda.', creationDate: '20/20/20 18:00 Hs'},
+                {id: 0, editDate: null, username: 'Jhon Mark', content: 'This is the body for a comment post. asd asd asd asd sad asd as das das dasda sdasd asdasdasd asdasd asdas dasdasd asda sdas dasdasd asdas dasd asd as dasd asd asda.', creationDate: '20/20/20 18:00 Hs'},
+                {id: 0, editDate: null, username: 'Jhon Mark', content: 'This is the body for a comment post. asd asd asd asd sad asd as das das dasda sdasd asdasdasd asdasd asdas dasdasd asda sdas dasdasd asdas dasd asd as dasd asd asda.', creationDate: '20/20/20 18:00 Hs'},
+                {id: 0, editDate: null, username: 'Jhon Mark', content: 'This is the body for a comment post. asd asd asd asd sad asd as das das dasda sdasd asdasdasd asdasd asdas dasdasd asda sdas dasdasd asdas dasd asd as dasd asd asda.', creationDate: '20/20/20 18:00 Hs'},
+                {id: 0, editDate: null, username: 'Jhon Mark', content: 'This is the body for a comment post. asd asd asd asd sad asd as das das dasda sdasd asdasdasd asdasd asdas dasdasd asda sdas dasdasd asdas dasd asd as dasd asd asda.', creationDate: '20/20/20 18:00 Hs'}
             ]
         }
     };
 
+    componentDidMount() {
+        this.props.location.state ?
+            this.setState({post: this.props.location.state.post, topic: this.props.location.state.topic}) :
+            this.getViewData()
+    }
+
+    getViewData = () => {
+        getPostData(this.props.match.params.id)
+            .then((res) => {
+                this.setState({post: res})
+                // with get topic by id endpoint
+                /*getTopicData(res.topicId)
+                    .then((res) => this.setState({topic: res}))
+                    .catch((err) => this.setState({topicErrorMessage: err}))*/
+            })
+            .catch((err) => this.setState({postErrorMessage: err}))
+    }
+
     handleCancel = () => {
         this.setState({showModal: false})
     }
 
     handleConfirm = () => {
-        deletePost(this.props.location.state.post.id).then(() =>
-            this.props.history.push('/main/topic', {topic : this.props.location.state.topic}))
+        deletePost(this.props.location.state ? this.props.location.state.post.id : this.props.match.params.id)
+            .then(() => this.props.history.push('/main/topic', {topic: this.state.topic}))
     }
 
     handleDelete = () => {
         this.setState({showModal: true})
     }
 
+    handleTopicNavigation = () => {
+        this.props.history.push('/main/topic/' + this.state.topic.id, {topic : this.state.topic})
+    }
+
```

Comment:
Unused method, either delete it or use it for the above correction (probably what was intended for) 😉